server/src/main/java/io/spine/server/commandbus/DuplicateCommandException.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2018, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.commandbus;
+
+import com.google.protobuf.Message;
+import io.spine.Identifier;
+import io.spine.base.Error;
+import io.spine.core.Command;
+import io.spine.core.CommandEnvelope;
+import io.spine.core.CommandValidationError;
+import io.spine.core.MessageInvalid;
+import io.spine.type.TypeName;
+
+import static java.lang.String.format;
+
+/**
+ * Reports an attempt to dispatch a duplicate command.
+ *
+ * <p>A command is considered a duplicate when its ID matches the ID of a command which was
+ * already dispatched emitting events in a target aggregate.
+ *
+ * @author Mykhailo Drachuk
+ */
+public class DuplicateCommandException extends CommandException implements MessageInvalid {
+
+    private static final long serialVersionUID = 0L;
+
+    private DuplicateCommandException(String messageText, Command command, Error error) {
+        super(messageText, command, error);
+    }
+
+    /**
+     * Creates an exception for a duplicate command dispatched to aggregate.
+     *
+     * @param command the duplicate command
+     * @return a newly constructed {@link DuplicateCommandException} instance
+     */
+    public static DuplicateCommandException forAggregate(Command command) {
```

Comment:
I don't like the naming, as you are passing an instance of `Command`, not an instance of `Aggregate`. That's why let's rename it to `of`.